modules/apps/27-interchain-accounts/controller/ibc_module_test.go

```diff
@@ -189,6 +190,54 @@ func (suite *InterchainAccountsTestSuite) TestOnChanOpenInit() {
 	}
 }
 
+// Test initiating a ChanOpenTry using the controller chain instead of the host chain
+// ChainA is the controller chain. ChainB creates a controller port as well,
+// attempting to trick chainA.
+// Sending a MsgChanOpenTry will never reach the application callback due to
+// core IBC checks not passing, so a call to the application callback is also
+// done directly.
+func (suite *InterchainAccountsTestSuite) TestChanOpenTry() {
+	suite.SetupTest() // reset
+	path := NewICAPath(suite.chainA, suite.chainB)
+	suite.coordinator.SetupConnections(path)
+
+	err := InitInterchainAccount(path.EndpointA, TestOwnerAddress)
+	suite.Require().NoError(err)
+
+	// chainB also creates a controller port
+	err = InitInterchainAccount(path.EndpointB, TestOwnerAddress)
+	suite.Require().NoError(err)
+
+	path.EndpointA.UpdateClient()
+	channelKey := host.ChannelKey(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+	proofInit, proofHeight := path.EndpointB.Chain.QueryProof(channelKey)
```

Comment:
So this `QueryProof` on the testchain just calls into ABCI for a proof that this channelKey is set in the store? 
Do I understand that correctly? 

What happens if it doesn't exist? Do you get an empty `[]byte` and height of -1 or something? 

Just asking out of my own curiosity :)

Reply:
> So this QueryProof on the testchain just calls into ABCI for a proof that this channelKey is set in the store?
Do I understand that correctly?

Correct! That the key is set in store with the expected value

> What happens if it doesn't exist? Do you get an empty []byte and height of -1 or something?

It returns a non existence proof

Reply:
Ah okay, great! So if you were to use the non existence proof then as argument to some other call, an error would eventually be propagated back to you I assume?

Reply:
correct, the non existence proof should be rejected by the tendermint client